includes/class-cli-commands.php

```diff
@@ -134,7 +140,12 @@ public function insert_redirect( $args, $assoc_args ) {
 			$to = esc_url_raw( $args[1] );
 		}
 
-		$redirect = Utilities\insert_redirect( $from, $to, $code );
+		$preserve_parameters = filter_var( $assoc_args['preserve-parameters'], FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE );
+		if ( $preserve_parameters === null || ! is_bool( $preserve_parameters ) ){
```

Comment:
So, today I learned/remembered about `filter_var` 😁. 
If you are passing `FILTER_VALIDATE_BOOLEAN` then you are going to get a bool or null,. so you don't need the `is_bool` in the test.